apps/hash-frontend/src/components/hooks/use-users-with-links.ts

```diff
@@ -0,0 +1,83 @@
+import { ApolloQueryResult, useQuery } from "@apollo/client";
+import { UserProperties } from "@local/hash-isomorphic-utils/system-types/shared";
+import {
+  AccountId,
+  Entity,
+  EntityRootType,
+  Subgraph,
+} from "@local/hash-subgraph";
+import { getRoots } from "@local/hash-subgraph/stdlib";
+import { useMemo } from "react";
+
+import {
+  QueryEntitiesQuery,
+  QueryEntitiesQueryVariables,
+} from "../../graphql/api-types.gen";
+import { queryEntitiesQuery } from "../../graphql/queries/knowledge/entity.queries";
+import { constructUser, User } from "../../lib/user-and-org";
+
+/**
+ * Retrieves a specific set of useranizations, with their avatars and members populated
+ */
+export const useUsersWithLinks = ({
+  userAccountIds,
+}: {
+  userAccountIds?: AccountId[];
+}): {
+  loading: boolean;
+  users?: User[];
+  refetch: () => Promise<ApolloQueryResult<QueryEntitiesQuery>>;
+} => {
+  const { data, loading, refetch } = useQuery<
+    QueryEntitiesQuery,
+    QueryEntitiesQueryVariables
+  >(queryEntitiesQuery, {
+    variables: {
+      operation: {
+        multiFilter: {
+          filters:
+            userAccountIds?.map((accountId) => ({
+              field: ["metadata", "recordId", "uuid"],
+              operator: "EQUALS",
+              value: accountId,
+            })) ?? [],
+          operator: "OR",
+        },
+      },
+      constrainsValuesOn: { outgoing: 0 },
+      constrainsPropertiesOn: { outgoing: 0 },
+      constrainsLinksOn: { outgoing: 0 },
+      constrainsLinkDestinationsOn: { outgoing: 0 },
+      inheritsFrom: { outgoing: 0 },
+      isOfType: { outgoing: 0 },
+      // These depths are chosen to cover the following:
+      // 1. the user's avatar (user -> [hasLeftEntity incoming 1] hasAvatar [hasRightEntity outgoing 1] -> avatar)
+      // 2. the user's user memberships (user <- [hasLeftEntity outgoing 1] userMembership [hasRightEntity incoming 1] <- user)
+      hasLeftEntity: { incoming: 1, outgoing: 1 },
+      hasRightEntity: { incoming: 1, outgoing: 1 },
```

Comment:
We aren't using the user's org memberships so I'd remove fetching it until we need it

Reply:
Good point, have addressed in https://github.com/hashintel/hash/pull/3392/commits/750938dadd3a2632adf054bea797b6bdc4129b63

Reply:
Also noticed we previously didn't make avatars public when they are uploaded, fixed in https://github.com/hashintel/hash/pull/3392/commits/fde36a03b4003b22e64e5f0199bffacb706f73c7